### PR TITLE
dependabot-cli: 1.41.0 -> 1.57.0, pin image dependencies

### PR DESCRIPTION
**Pull request by infinisil**
Noticed that the dependabot updates for https://github.com/NixOS/nixpkgs-vet aren't working anymore. Tracked it down to it needing a dependabot CLI update, but to prevent future breakage, this PR also introduces a `dependabot-pinned` wrapper that pins the docker images.

Ping @NixOS/nixpkgs-vet 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Successfully ran the nixpkgs-vet update script making use of dependabot

---

This work is sponsored by [Antithesis](https://antithesis.com/) :sparkles:

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by philiptaron — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352866`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dependabot-cli</li>
  </ul>
</details>

**Comment by philiptaron** [comment]
```
$ results/dependabot-cli-x86_64-linux/bin/dependabot-pinned --help
results/dependabot-cli-x86_64-linux/bin/dependabot-pinned: line 2: docker: command not found
```

**Comment by infinisil** [comment]
Ah yeah that's fair, I guess let's merge this as is for now, but consider doing that in the future if this becomes more complex, or people demand smaller closure sizes :)

**Comment by infinisil** [comment]
Requested a review from the maintainer manually because ofborg was apparently too slow. Let us know if you have any input @l0b0, happy to address retroactively!

File: pkgs/by-name/de/dependabot-cli/package.nix [hunk 1]
```diff
@@ -5,10 +5,37 @@
   installShellFiles,
   lib,
   testers,
+  dockerTools,
+  makeWrapper,
```
Comment (l0b0) [inline pkgs/by-name/de/dependabot-cli/package.nix:9]: Nit: These should be sorted alphabetically with the rest.

File: pkgs/by-name/de/dependabot-cli/package.nix [hunk 2]
```diff
@@ -16,28 +44,38 @@ buildGoModule {
...
       --bash <($out/bin/dependabot completion bash) \
       --fish <($out/bin/dependabot completion fish) \
       --zsh <($out/bin/dependabot completion zsh)
+
+    # Create a wrapper that pins the docker images that are depended upon
+    makeWrapper $out/bin/dependabot $out/bin/dependabot-pinned \
+      --run "docker load --input ${updateJobProxy}" \
+      --add-flags "--proxy-image=dependabot-update-job-proxy:${tag}" \
+      --run "docker load --input ${updaterGitHubActions}" \
+      --add-flags "--updater-image=dependabot-updater-github-actions:${tag}"
```
Comment (l0b0) [inline pkgs/by-name/de/dependabot-cli/package.nix:74]: Could this be a [separate output](https://github.com/NixOS/nixpkgs/pull/352866#issuecomment-2452142605), or behind a feature flag like `withDockerImages`?

**Review by l0b0 — COMMENTED** [review]
Thank you!

**Comment by infinisil** [comment]
https://github.com/NixOS/nixpkgs/pull/354085

**Pull request merged**